### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by DilsherB**
In this PR:
1. initial state changed from empty to something.
2. fetching the data from slice.
3. remving the book by using redux

### CODE REVIEW

File: src/components/BookDetail.js [hunk 1]
```diff
@@ -1,71 +1,31 @@
...
-    },
-  ];
-  const [books, setBooks] = useState(booksDB);
-  const handleDeleteBook = (id) => {
-    setBooks((oldBooks) => {
-      return oldBooks.filter((theBook) => theBook.id !== id);
-    });
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  // const [books, setBooks] = useState(booksDB);
```
Comment (rustam-tolipov) [inline src/components/BookDetail.js]: **Great job!**

- [ ] Books is loading as expected. However, your remove functionality of your app is not working. It's showing an error. Please fix the issue and make sure app is running as expected!
 Screenshot 2023-06-13 at 18 49 19

Comment (rustam-tolipov) [inline src/components/BookDetail.js]: - [x] Kindly remove unused commented code from your file!

File: src/components/BookDetail.js [hunk 2]
```diff
@@ -1,71 +1,31 @@
...
+  const handleDeleteBook = ({ id }) => {
+    // const { id } = e.target.dataset.id;
```
Comment (rustam-tolipov) [inline src/components/BookDetail.js]: - [x] Kindly remove unused commented code from your file!

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -2,7 +2,50 @@ import { createSlice } from "@reduxjs/toolkit";
 
 export const booksSlice = createSlice({
   name: "book",
-  initialState: [],
+  initialState: [
+    {
+      item_id: "item1",
```
Comment (rustam-tolipov) [inline src/redux/books/booksSlice.js]: [OPTIONAL] I think it woud be great idea to use `id: 1`, number insted of `item_id: "item1"`

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @DilsherB,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Earned a star for a good readme_
_Most of the functionality has been implemented_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookDetail.js (hunk 1 above)
Comment (DilsherB) [reply src/components/BookDetail.js]: @Rustamxon7
Thank you for your detailed review feed back. 
I tried a much to implement remove functionality but couldn't. Please help me if you can.

Comment (DilsherB) [reply src/components/BookDetail.js]: removed

File: src/components/BookDetail.js (hunk 2 above)
Comment (DilsherB) [reply src/components/BookDetail.js]: removed

File: src/redux/books/booksSlice.js (hunk 3 above)
Comment (DilsherB) [reply src/redux/books/booksSlice.js]: done

File: src/components/BookDetail.js (hunk 1 above)
Comment (DilsherB) [reply src/components/BookDetail.js]: solved.

File: src/components/NewBookForm.js [hunk 4]
```diff
@@ -7,20 +7,22 @@ const NewBookForm = () => {
...
+              name="category"
+              id="category"
+              className="form-select form-control"
+            >
+              <option value="Action">Action</option>
+              <option value="Science Fiction">Science Fiction</option>
+              <option value="Economy">Economy</option>
+            </select>
           </div>
           <div className="form-group col-2 col-sm-3">
```
Comment (topeogunleye) [inline src/components/NewBookForm.js:27]: - [ ] [OPTIONAL] As this was not mentioned by the previous reviewer. I kindly suggest that you ensure that your Adding of books is possible. see video below for more information:

https://www.loom.com/share/a431c46f59924682831af5d2a78c720b?sid=cdae3d52-e51e-4856-9891-dab6e449a367

**Review by topeogunleye — APPROVED** [review]
### Hi @DilsherB, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
-  Fixed removing a book functionality as suggested. ✔️ 
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
* [ ] For future pull requests, please notice that your branch name `redux` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-redux`. This makes it easier to track the changes and the purpose of each branch.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**